src/file-upload/file-list.tsx

```diff
@@ -321,14 +500,43 @@ export const FileList = ({
         });
     };
 
-    if (!fileItems || fileItems.length === 0) return null;
+    /**
+     * Renders the progress status of each file upload.
+     * In progress: 25% complete for uploading filename.pdf
+     * Complete: filename.pdf upload is complete
+     * Error: Error uploading filename.pdf: error message
+     *
+     * @returns A visually hidden element announcing the progress status.
+     */
+    const renderProgressStatus = () => (
+        <VisuallyHidden
+            ref={visuallyHiddenRef}
+            tabIndex={-1}
+            aria-live="polite"
+            aria-atomic="true"
+        >
+            {progressAnnouncement}
+        </VisuallyHidden>
+    );
 
-    if (disabled || readOnly || !shouldEnableSort()) {
+    const renderItemsWithWrapper = () => {
         return (
-            <ListWrapper $readOnly={readOnly} ref={wrapperRef}>
-                {renderItems()}
-            </ListWrapper>
+            <>
+                {renderProgressStatus()}
+                <ListWrapper
+                    tabIndex={0}
```

Comment:
does this specifically need to be in the tab order? or is setting `-1` to keep it focusable sufficient?